pkg/cluster/cluster_config.go

```diff
@@ -287,3 +288,32 @@ func IsDefaultAuthMethod(ctx context.Context, cli client.Client) (bool, error) {
 	// we only create userGroups for "IntegratedOAuth" or "" and leave other or new supported type value in the future
 	return authenticationobj.Spec.Type == configv1.AuthenticationTypeIntegratedOAuth || authenticationobj.Spec.Type == "", nil
 }
+
+func GetNodeArchitectures(ctx context.Context, client client.Client) (map[string]struct{}, error) {
+	nodeList := &corev1.NodeList{}
+	if err := client.List(ctx, nodeList); err != nil {
```

Comment:
maybe we only need to list worker nodes, for an hosted control plane based cluster, the control plane nodes may be different from the worker nodes.

Reply:
I agree we only should care about the worker nodes. Fixed in the latest commit.